pkg/util/kubernetes/autoscalers/processor.go

```diff
@@ -146,13 +151,25 @@ func (p *Processor) ProcessWPAs(wpa *v1alpha1.WatermarkPodAutoscaler) map[string
 }
 
 func makeChunks(batch []string) (chunks [][]string) {
-	for i := 0; i < len(batch); i += chunkSize {
-		if i+chunkSize > len(batch) {
-			chunks = append(chunks, batch[i:])
-			break
+	// uriLength is used to avoid making a query that goes beyond the maximum URI size.
+	var uriLength int
+	var tempBucket []string
+
+	for _, val := range batch {
+		// Length of the query plus comma, time and space aggregators that come later on.
+		tempSize := len(url.PathEscape(val)) + extraQueryCharacters
```

Comment:
```suggestion
		tempSize := len(url.QueryEscape(val)) + extraQueryCharacters
```

Reply:
Okay for this one, although the characters they differentiate are not going to be in the queries.